### ISSUE TITLE
[4.0] Class caps naming

### ISSUE THREAD

**Issue by wilsonge** [joomla/joomla-cms#27014]
### Steps to reproduce the issue
View appveyor output

```
[5.5MiB/106.12s] Deprecation Notice: Class Joomla\Tests\Unit\Libraries\Cms\Table\TableTest located in C:/projects/joomla-cms/tests\Integration\Libraries\Cms\Table\TableTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v1.11+. in phar://C:/Tools/php/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:185
[5.5MiB/106.13s] Deprecation Notice: Class Joomla\Tests\Unit\Libraries\Cms\Html\DocumentTest located in C:/projects/joomla-cms/tests\Unit\Libraries\Cms\Document\DocumentTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v1.11+. in phar://C:/Tools/php/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:185
[5.6MiB/106.45s] Deprecation Notice: Class Joomla\CMS\Filesystem\Support\StringController located in C:/projects/joomla-cms/libraries/src\Filesystem\Support\Stringcontroller.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v1.11+. in phar://C:/Tools/php/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:185
[5.7MiB/106.51s] Deprecation Notice: Class Joomla\CMS\Form\Rule\SubformRule located in C:/projects/joomla-cms/libraries/src\Form\Rule\SubFormRule.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v1.11+. in phar://C:/Tools/php/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:185
[5.7MiB/106.52s] Deprecation Notice: Class Joomla\CMS\Filter\Image\Sketchy located in C:/projects/joomla-cms/libraries/src\Image\Filter\Sketchy.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v1.11+. in phar://C:/Tools/php/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:185
```

These files/class names are all not case matching. We need to fix this to work on future versions of composer (and probably already cause issues in Linux vs Windows). This may also affect Joomla 3.x @HLeithner

**Comment by brianteeman** [comment]
Just the two unit tests to fix after #27015 and #27030

**Comment by brianteeman** [comment]
I believe that #27464 addresses the final parts of this - if wrong it can always be reopened
